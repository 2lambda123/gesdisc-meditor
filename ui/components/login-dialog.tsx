import Modal from 'react-bootstrap/Modal'
import Button from 'react-bootstrap/Button'
import { MdPerson } from 'react-icons/md'
import styles from './login-dialog.module.css'

const LoginDialog = ({
    show = false,
    onHide = () => {},
}) => {
    function login() {
        window.location.href = '/meditor/api/login'
    }

    return (
        <Modal show={show} centered onHide={onHide} animation={false}>
            <Modal.Body className={styles.body}>
                <h3>Welcome!</h3>

                mEditor requires that you be an authorized user to add models or edit documents, so please...

                <Button onClick={login} variant="link">
                    <MdPerson />
                    Login
                </Button>

                <h5>No account? Please <a href="https://urs.earthdata.nasa.gov">register</a></h5>
            </Modal.Body>
        </Modal>
    )
}

export default LoginDialog
