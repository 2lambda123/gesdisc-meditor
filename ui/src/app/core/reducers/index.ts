import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
