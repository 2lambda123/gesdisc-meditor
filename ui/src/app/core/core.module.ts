import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
