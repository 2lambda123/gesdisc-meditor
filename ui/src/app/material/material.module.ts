import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
