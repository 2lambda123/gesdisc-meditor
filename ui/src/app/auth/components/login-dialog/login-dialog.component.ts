import { Component } from '@angular/core';
import { MatDialogRef } from '@angular/material';

import { environment } from '../../../../environments/environment';

@Component({
	selector: 'med-login-dialog',
	template: `
		<h1 mat-dialog-title>Welcome!</h1>
		<mat-dialog-content>
			The Model Editor requires that you
			be an authorized user to add models
			or edit documents, so please...
			<button mat-button (click)="login()" color="accent" class="login-btn">
				<mat-icon>person</mat-icon>
				Login
			</button>
		</mat-dialog-content>
		<h5> No account? Please <a href="https://urs.earthdata.nasa.gov">register</a></h5>`,
	styles: [
		`
			h1, h5, mat-dialog-content {
				text-align: center;
			}

			h1 {
				font-size: 24px;
			}

			.login-btn {
				margin: 10px 0;
			}
		`
	]
})

export class LoginDialog {

	constructor(
		public dialogRef: MatDialogRef<LoginDialog>
	) {}


	login() {
		window.location.href = this.getApiUrl() + '/login';
	}

	getApiUrl() {
		const basePath = environment.API_BASE_PATH;
		return basePath.indexOf('http') !== 0 ? window.location.origin + basePath : basePath;
	}

}
