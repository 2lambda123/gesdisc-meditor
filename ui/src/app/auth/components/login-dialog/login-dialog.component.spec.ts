import { TestBed, ComponentFixture } from '@angular/core/testing';
import { MatIconModule, MatDialogModule, MatDialogRef } from '@angular/material';
import { LoginDialog } from './login-dialog.component';

describe('Login Component', () => {
	let fixture: ComponentFixture<LoginDialog>;
	let component: LoginDialog;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				MatIconModule,
				MatDialogModule
			],
			declarations: [ LoginDialog ],
			providers: [
				{ provide: MatDialogRef, useValue: {} }
			]
		});

		fixture = TestBed.createComponent(LoginDialog);
		component = fixture.componentInstance;

		fixture.detectChanges();
	});

	it('should compile login dialog', () => {
		expect(fixture).toMatchSnapshot();
	});
});
