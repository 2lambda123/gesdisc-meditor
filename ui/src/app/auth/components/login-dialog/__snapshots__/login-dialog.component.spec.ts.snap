// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should compile login dialog 1`] = `
<med-login-dialog
  dialogRef={[Function Object]}
>
  <h1
    _ngcontent-c0=""
    class="mat-dialog-title"
    id="mat-dialog-title-0"
    mat-dialog-title=""
  >
    Welcome!
  </h1><mat-dialog-content
    _ngcontent-c0=""
    class="mat-dialog-content"
  >
     The Model Editor requires that you be an authorized user to add models or edit documents, so please... 
    <button
      _ngcontent-c0=""
      class="login-btn"
      color="accent"
      mat-button=""
    >
      <mat-icon
        _ngcontent-c0=""
        aria-hidden="true"
        class="mat-icon material-icons"
        role="img"
      >
        person
      </mat-icon>
       Login 
    </button>
  </mat-dialog-content><h5
    _ngcontent-c0=""
  >
     No account? Please 
    <a
      _ngcontent-c0=""
      href="https://urs.earthdata.nasa.gov"
    >
      register
    </a>
  </h5>
</med-login-dialog>
`;
